pyproject.toml

```diff
@@ -1,7 +1,97 @@
 [build-system]
-requires = ["jupyter_packaging~=0.7.9", "jupyterlab>=3,<=4", "setuptools>=40.8.0", "wheel"]
-build-backend = "setuptools.build_meta"
+requires = ["hatchling>=1.5.0", "hatch-nodejs-version"]
+build-backend = "hatchling.build"
+
+[project]
+name = "jupytext"
+description = "Jupyter notebooks as Markdown documents, Julia, Python or R scripts"
+license = { file = "LICENSE" }
+authors = [
+    { name = "Marc Wouts", email = "marc.wouts@gmail.com" },
+]
+readme = "README.md"
+requires-python = ">=3.8"
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "License :: OSI Approved :: MIT License",
+    "Environment :: Console",
+    "Framework :: Jupyter",
+    "Intended Audience :: Developers",
+    "Intended Audience :: Science/Research",
+    "Topic :: Text Processing :: Markup",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+]
+dependencies = [
+    "nbformat",
+    "pyyaml",
+    "toml",
+    "markdown-it-py>=1.0.0",
+    "mdit_py_plugins",
+]
+dynamic = ["version"]
+
+[project.urls]
+homepage = "https://github.com/mwouts/jupytext"
+source = "https://github.com/mwouts/jupytext"
+documentation = "https://jupytext.readthedocs.io/en/latest/index.html"
+
+[project.optional-dependencies]
+extension = [
+    "jupytext_extension",
+]
+dev = [
+    "autopep8",
+    "black",
+    "isort",
+    "flake8",
+    "pytest",
+    "pytest-cov>=2.6.1",
+    "gitpython",
+    "jupyterlab",
+    "notebook",
+    "nbconvert",
+    "ipykernel",
+]
```

Comment:
I prefer to have these separated at least as:
- `test` used by all downstream to run tests
- `docs` used by RTD
- `test-cov` used by the CI to add the coverege
- `dev` used by the developers to add `mypy`, pre-commit`, etc.